AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/AWSS3StorageServiceBehavior.swift

```diff
@@ -9,6 +9,7 @@ import Foundation
 import Amplify
 import AWSS3
 
+typealias AWSS3StorageServiceProvider = () throws -> AWSS3StorageServiceBehavior
```

Comment:
I guess the answer depends on what you consider a delegate vs a provider :P 
I went with provider because it's just a callback that gets invoked to retrieve an actual `AWSS3StorageServiceBehavior` instance.
I didn't want to go with a full protocol because then we'd have to deal with `weak`s  and unwrappings.

We could name it `AWSS3StorageServiceRetriever` if that reads better.